Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Models/Attributes/TemplateParametersExtension.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Amazon.Lambda.Annotations.SourceGenerator.Models.Attributes
+{
+    public static class TemplateParametersExtension
+    {
+        /// <summary>
+        /// Returns aa <see cref="HashSet{T}"/> of string containing template parameter
+        /// parsed from the <see cref="RestApiAttribute.Template"/>.
+        /// </summary>
+        /// <param name="attribute">this <see cref="RestApiAttribute"/> object.</param>
+        public static HashSet<string> GetTemplateParameters(this RestApiAttribute attribute)
+        {
+            var template = attribute.Template;
+            return GetTemplateParameters(template);
+        }
+
+        /// <summary>
+        /// Returns aa <see cref="HashSet{T}"/> of string containing template parameter
+        /// parsed from the <see cref="HttpApiAttribute.Template"/>.
+        /// </summary>
+        /// <param name="attribute">this <see cref="HttpApiAttribute"/> object.</param>
+        public static HashSet<string> GetTemplateParameters(this HttpApiAttribute attribute)
+        {
+            var template = attribute.Template;
+            return GetTemplateParameters(template);
+        }
+
+        private static HashSet<string> GetTemplateParameters(string attribute)
+        {
+            var parameters = new HashSet<string>();
+
+            var components = attribute.Split('/').Where(c => !string.IsNullOrWhiteSpace(c));
+            foreach (var component in components)
+            {
+                if (component.StartsWith("{") && component.EndsWith("}"))
+                {
+                    var parameter = component.Substring(1, component.Length - 2);
+                    if (IsValid(parameter))
```

Comment:
If there is a `FromRoute` attribute that is mapping the template parameter name to the C# variable name does the template parameter name have to follow C# naming rules?

Reply:
Also it doesn't every seem valid to skip parameters. If we can't handle the name then we should throw an error.

Reply:
> If there is a FromRoute attribute that is mapping the template parameter name to the C# variable name does the template parameter name have to follow C# naming rules?

No, atleast in ASP.NET Core, the logic needs to modified to accomodate FromRoute attribute with any string. I just tested 
```
        [HttpGet("{1}/{id}")]
        public string Get([FromRoute(Name = "1")]int id2, int id)
        {
            return "value";
        }
```
and it works.

> Also it doesn't every seem valid to skip parameters. If we can't handle the name then we should throw an error.

That's fair, I'm okay throwing an error and let customer know that need to fix the template.

Reply:
I just removed the validation logic all together.

Reply:
Added route parameter validation, now generator makes sure every route parameter is mapped. If any parameter found unmapped, it throws.